core/adapters/sqldatawarehouse.ts

```diff
@@ -0,0 +1,113 @@
+import { Adapter } from "@dataform/core/adapters/base";
+import { IAdapter } from "@dataform/core/adapters/index";
+import { Task, Tasks } from "@dataform/core/tasks";
+import { dataform } from "@dataform/protos";
+
+export class SQLDataWarehouseAdapter extends Adapter implements IAdapter {
+  public resolveTarget(target: dataform.ITarget) {
+    return `"${target.schema}"."${target.name}"`;
+  }
+
+  public publishTasks(
+    table: dataform.ITable,
+    runConfig: dataform.IRunConfig,
+    tableMetadata: dataform.ITableMetadata
+  ): Tasks {
+    const tasks = Tasks.create();
+    // Drop the existing view or table if we are changing its type.
+    if (tableMetadata && tableMetadata.type !== this.baseTableType(table.type)) {
+      tasks.add(
+        Task.statement(this.dropIfExists(table.target, this.oppositeTableType(table.type)))
+      );
+    }
+    if (table.type === "incremental") {
+      if (runConfig.fullRefresh || !tableMetadata || tableMetadata.type === "view") {
+        tasks.addAll(this.createOrReplace(table));
+      } else {
+        // The table exists, insert new rows.
+        tasks.add(
+          Task.statement(
+            this.insertInto(
+              table.target,
+              tableMetadata.fields.map(f => f.name),
+              this.where(table.query, table.where)
+            )
+          )
+        );
+      }
+    } else {
+      tasks.addAll(this.createOrReplace(table));
+    }
+    return tasks;
+  }
+
+  public assertTasks(
+    assertion: dataform.IAssertion,
+    projectConfig: dataform.IProjectConfig
+  ): Tasks {
+    const target =
+      assertion.target ||
+      dataform.Target.create({
+        schema: projectConfig.assertionSchema,
+        name: assertion.name
+      });
+
+    return Tasks.create()
+      .add(Task.statement(this.dropIfExists(target, "view")))
+      .add(
+        Task.statement(`
+        create view ${this.resolveTarget(target)}
+        as ${assertion.query}`)
+      )
+      .add(Task.assertion(`select sum(1) as row_count from ${this.resolveTarget(target)}`));
+  }
+
+  public dropIfExists(target: dataform.ITarget, type: string) {
+    if (type === "view") {
+      return `drop ${this.baseTableType(type)} if exists ${this.resolveTarget(target)} `;
+    }
+    return `if object_id ('${this.resolveTarget(
+      target
+    )}','U') is not null drop table ${this.resolveTarget(target)}`;
+  }
+
+  private createOrReplace(table: dataform.ITable) {
+    if (table.type === "view") {
+      return (
+        Tasks.create()
+          // Drop the view in case we are changing the number of column(s) (or their types).
+          .add(Task.statement(this.dropIfExists(table.target, this.baseTableType(table.type))))
```

Comment:
It would be much nicer to user `alter view` if we can. I did this for redshift because there was no other way, but this increases downtime for consumers while we run.

I think this is as simple as changing `create` to `alter` when the view exists, i just did a test manually and it seems to work.